### ISSUE TITLE
Invalid assignment of ObjectType in saved_query list API

### ISSUE THREAD

**Issue by akzarma** [apache/superset#23610]
There is wrong assignment of object_type [here](https://github.com/apache/superset/blob/master/superset/models/sql_lab.py#LL374C1-L375C1)
Due to which, saved_query list API fails to load (when used databases except on sqlite - probably because sqlite does not strictly check enum type of object_type)

#### How to reproduce the bug

1. Setup backend database  as postgres, Go to 'Saved Queries' page
2. You'll see an error popup right away and there are no saved queries shown.

### Expected results

There should not be any error and saved queries should load fine
### Actual results

- Shows error popup with sql error and there is no row of saved queries.

#### Screenshots

<img width="1509" alt="Screenshot 2023-04-06 at 4 51 32 PM" src="https://user-images.githubusercontent.com/25038017/230376783-c2a7efbf-b004-4cf9-b939-f06cd200910f.png">

### Environment

(please complete the following information):

browser type and version:
superset version: 2.1.0
python version: 3.9.14
node.js version: 16.17.1
any feature flags active: TAGGING_SYSTEM

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ ] I have reproduced the issue with at least the latest released version of superset.
- [ ] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by akzarma** [comment]
@Antonio-RiveroMartnez what database is your local setup running on? it works fine for SQLite. I can repro it with postgres.

**Comment by Antonio-RiveroMartnez** [comment]
Yes, I am using Postgres (You can verify in the video above for the selected Database). Are you using latest master or branch 2.1.0? If 2.1.0, could you try using latest master? Trying to see if it got already solved by some newer change after the 2.1.0 cut. Thanks @akzarma

**Comment by akzarma** [comment]
Yeah, my bad, I was some commits behind. it seems like it got resolved in the effect of https://github.com/apache/superset/pull/23526
However, The fix (#23611 ) I provided is still needed for the Tag feature to work on saved queries (which I think is half baked for saved queries)
My fix assigns the right object_type ("query") from the available types instead of "saved_query".
https://github.com/apache/superset/blob/b820eb82353fedf92aa45f1293ac3dcf4020296d/superset/tags/models.py#L66-L74 
But yeah, this issue should be closed as the steps mentioned are not reproducing the bug.
Thanks @Antonio-RiveroMartnez
